src/colmap/estimators/bundle_adjustment.cc

```diff
@@ -825,6 +828,189 @@ void RigBundleAdjuster::ParameterizeCameraRigs(Reconstruction* reconstruction) {
+  std::vector<double> vini_err2_wrt_prior;
+  std::vector<Eigen::Vector3d> v_src, v_dst;
+  vini_err2_wrt_prior.reserve(config_.NumImages());
+  v_src.reserve(config_.NumImages());
+  v_dst.reserve(config_.NumImages());
+  for (const auto& image_id : config_.Images()) {
+    const auto& image = reconstruction->Image(image_id);
+    if (image.HasPosePrior()) {
+      v_src.push_back(image.ProjectionCenter());
+      v_dst.push_back(image.WorldFromCamPrior().position);
+      vini_err2_wrt_prior.push_back(
+          (v_src.back() - v_dst.back()).squaredNorm());
+    }
+  }
+
+  // Fallback to classic BA if not enough registered images with a prior
+  // position
+  prior_options_.use_prior_position = v_src.size() > 2;
+
+  // Fix 7-DOFs of BA problem for classic BA
+  if (!prior_options_.use_prior_position) {
+    const std::vector<image_t>& reg_image_ids = reconstruction->RegImageIds();
+    config_.SetConstantCamPose(reg_image_ids[0]);
+    config_.SetConstantCamPositions(reg_image_ids[1], {0});
+  }
+
+  // Apply RANSAC-based Sim3 Alignment
+  if (prior_options_.use_prior_position) {
+    LOG(INFO) << "Initial alignment error w.r.t. prior position:";
+    LOG(INFO) << "  - rmse:   " << std::sqrt(Mean(vini_err2_wrt_prior)) << " m";
+    LOG(INFO) << "  - median: " << std::sqrt(Median(vini_err2_wrt_prior))
+              << " m";
+
+    RANSACOptions ransac_options;
+    ransac_options.max_error = (options_.prior_position_std * 3.).norm();
+
+    LORANSAC<SimilarityTransformEstimator<3, true>,
+             SimilarityTransformEstimator<3, true>>
+        ransac(ransac_options);
+
+    const auto report = ransac.Estimate(v_src, v_dst);
+
+    if (report.success) {
+      // Apply sim3 transform
+      const Sim3d tform = Sim3d::FromMatrix(report.model);
+      reconstruction->Transform(tform);
+
+      std::vector<double> verr2_wrt_prior;
+      verr2_wrt_prior.reserve(vini_err2_wrt_prior.size());
+      for (const auto& image_id : config_.Images()) {
+        const auto& image = reconstruction->Image(image_id);
+        if (image.HasPosePrior()) {
+          verr2_wrt_prior.push_back(
+              (image.ProjectionCenter() - image.WorldFromCamPrior().position)
+                  .squaredNorm());
+        }
+      }
+
+      LOG(INFO) << "Rigid Sim3 alignment w.r.t. prior position:";
+      LOG(INFO) << "  - scale : " << tform.scale;
+      LOG(INFO) << "  - trans : " << tform.translation.transpose();
+      // LOG(INFO) << "  - rot : " << tform.rotation.coeffs().transpose();
+
+      LOG(INFO) << "Sim3 alignment error w.r.t. prior position:";
+      LOG(INFO) << "  - rmse:   " << std::sqrt(Mean(verr2_wrt_prior)) << " m";
+      LOG(INFO) << "  - median: " << std::sqrt(Median(verr2_wrt_prior)) << " m";
+
+    } else {
+      LOG(WARNING) << "Sim3 alignment w.r.t. prior position failed!";
+    }
+  }
+
+  // Put the centroid of the reconstruction to (0.,0.,0.)
+  // to avoid any numerical instability
+  Sim3d sim_to_center;
```

Comment:
Reconstruction::Normalize() could achieve the same but maybe a bit more robust in case of outliers. What do you think about changing this method to the following interface:
```
  // Normalize scene by scaling and translation to avoid degenerate
  // visualization after bundle adjustment and to improve numerical
  // stability of algorithms.
  //
  // Translates scene such that the mean of the camera centers or point
  // locations are at the origin of the coordinate system.
  //
  // Scales scene such that the minimum and maximum camera centers are at the
  // given `extent`, whereas `p0` and `p1` determine the minimum and
  // maximum percentiles of the camera centers considered.
  Sim3d Normalize(bool fixed_scale = true,
                  double extent = 10.0,
                  double p0 = 0.1,
                  double p1 = 0.9,
                  bool use_images = true);
```

Reply:
Indeed something like this would work!  I have modified the PosePriorBundleAdjuster to hold the current sim3d transformation that has been applied to the reconstruction.  This actually led to a cleaner and more efficient code as the position priors are now transformed only when creating the cost function.  I think it would not be much work to set this sim3d object to the result of Normalize().  I am leaving it as it is for the moment due to @sarlinpe idea about storing a "gauge" object that would hold the transformation between the normalized and metric reconstruction.

Reply:
I suggest relying on `Reconstruction::Normalize` for now. I can look into adding the gauge object later - it requires changes beyond the scope of this PR.

Reply:
I have changed the Reconstruction::Normalize() method with the interface you suggested.  I have only set fixed_scale to false by default instead of true in order for the method to have the same behavior as it used to by default.  I am now relying on it in the PosePriorBundleAdjuster to normalize the reconstruction before solving the ceres problem and transforming it back to the metric coordinates after.

Reply:
Nice, thank you for all this great work!